repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/valkyrie/CullingBypassValkyrieAuthorizationTest.groovy

```diff
@@ -0,0 +1,247 @@
+        mc.orphanedHandlings.each {
+            e -> assert e.request.headers.contains("x-trans-id")
+        }
+
+
+
+        where:
+        method | tenantID       | deviceID | deviceID2 | permission     | responseCode | size
+        "GET"  | randomTenant() | "520707" | "511123"  | "view_product" | "200"        | 1
+        "GET"  | randomTenant() | "520708" | "511123"  | "view_product" | "200"        | 1
+        "GET"  | randomTenant() | "520707" | "520708"  | "view_product" | "200"        | 2
+        "GET"  | randomTenant() | "520705" | "520706"  | "view_product" | "200"        | 0
+    }
+
+    def "Test missing tenantid"() {
+        given: "a list permission devices defined in Valkyrie"
+        fakeIdentityV2Service.with {
+            client_token = UUID.randomUUID().toString()
+            client_tenantid = ""
+        }
+
+        fakeValkyrie.with {
+            device_id = "520707"
+            device_id2 = "520708"
+            device_perm = "view_product"
+        }
+
+        "Json Response from origin service"
+        def jsonResp = { request -> return new Response(200, "OK", ["content-type": "application/json"], jsonrespbody) }
+
+        when: "a request is made against a device with Valkyrie set permissions"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint + "/resources", method: "GET",
+                headers: [
+                        'content-type': 'application/json',
+                        'X-Auth-Token': fakeIdentityV2Service.client_token,
+                        'x-contact-id': '123456'
+                ],
+                defaultHandler: jsonResp
+        )
+
+        then:
+        mc.receivedResponse.code == "401"
+    }
+
+    // pre-authorized-roles have been config should be bypassed valkyrie authorization
+    // Just a pass through but not actual talk nor get any response from valkyrie
+    def "Bypass Valkyrie authorization on Culling"() {
+        given: "a list permission devices defined in Valkyrie token without tenantid associated with"
+        fakeIdentityV2Service.with {
+            client_token = "rackerSSO"
+            service_admin_role = "racker"
+        }
+
+        fakeValkyrie.with {
+            device_id = "520707"
+            device_id2 = "520708"
+            device_perm = "view_product"
+        }
+
+        "Json Response from origin service"
+        def jsonResp = { request -> return new Response(200, "OK", ["content-type": "application/json"], jsonrespbody) }
+
+        when: "a request is made against a device with Valkyrie set permissions"
+        MessageChain mc = deproxy.makeRequest(url: reposeEndpoint + "/resources", method: "GET",
+                headers: [
+                        'content-type': 'application/json',
+                        'X-Auth-Token': fakeIdentityV2Service.client_token,
+                ],
+                defaultHandler: jsonResp
+        )
+
+        then: "check response"
+        mc.receivedResponse.code == "200"
```

Comment:
Could we perhaps add an assertion that the fakeIdentityV2Service has not been interacted with?